codec-http2/src/main/java/io/netty/handler/codec/http2/Http2Connection.java

```diff
@@ -257,11 +256,22 @@
         int maxActiveStreams();
 
         /**
-         * Sets the maximum number of streams (created by this endpoint) that are allowed to be active at once.
-         * This is the {@code SETTINGS_MAX_CONCURRENT_STREAMS} value sent from the opposite endpoint to
+         * The limit imposed by {@link #maxActiveStreams()} does not apply to streams in the IDLE state. Since IDLE
+         * streams can still consume resources this limit will include streams in all states.
+         * @return The maximum number of streams that can exist at any given time.
+         */
+        int maxStreams();
+
+        /**
+         * Sets the limit for {@code SETTINGS_MAX_CONCURRENT_STREAMS} and the limit for {@link #maxStreams()}.
+         * @param maxActiveStreams The maximum number of streams (created by this endpoint) that are allowed to be
+         * active at once. This is the {@code SETTINGS_MAX_CONCURRENT_STREAMS} value sent from the opposite endpoint to
          * restrict stream creation by this endpoint.
+         * @param maxStreams The limit imposed by {@link #maxActiveStreams()} does not apply to streams in the IDLE
+         * state. Since IDLE streams can still consume resources this limit will include streams in all states.
+         * @throws Http2Exception if {@code maxStreams < maxActiveStream}.
          */
-        void maxActiveStreams(int maxActiveStreams);
+        void maxStreams(int maxActiveStreams, int maxStreams) throws Http2Exception;
```

Comment:
I debated this ... but then there is an awkward relationship between the two methods. `maxActiveStreams(..)` always has to check that the value is `<= maxStreams`. Having a single method emphasizes the relationship and the user can always pass explicitly `endpoint.maxStreams()` if they don't intend on changing it. WDYT?